lib/auth/exec.js

```diff
@@ -34,9 +35,13 @@ class ExecAuth {
       promise = Promise.resolve(this.auth_provider);
     } else {
       promise = new Promise((resolve, reject) => {
-        const env = {};
+        let env;
```

Comment:
Shouldn't it default to `process.env`, irrespectively of `this.auth_provider.env`?

Reply:
I think you would then modify the process.env Object by reference when adding values to env?

That shouldn't be a problem unless you were switching between contexts where environment variables would all end up in process.env and produce undesired behavior...

Reply:
Here I meant that env should be the concatenation of `process.env` and `this.auth_provider.env`. Hope that's clearer 😀.

Reply:
> I think you would then modify the process.env Object by reference when adding values to env?

You can do `let env = Object.assign({}, process.env, ...)`

Reply:
Done

Reply:
I just remembered why I didn't do it that way in the first place: we are now always copying process.env even when we don't need to

Reply:
I think that's acceptable. I prefer conciseness over performance in that case.